Caused by: java.lang.IllegalArgumentException: Invalid char ([) found at index (0) in sheet name '[select 1]'

When I export data from SQL Server using the .xlsx format, it raises an error (the error is described below). And I have tried SQL Server, pg (PostgreSQL), and MySQL; only SQL Server raises the error.

org.jkiss.dbeaver.model.exec.DBCException: Error while exporting table row
	at org.jkiss.dbeaver.tools.transfer.stream.StreamTransferConsumer.fetchRow(StreamTransferConsumer.java:247)
	at io.cloudbeaver.service.data.transfer.impl.WebServiceDataTransfer$3.fetchRow(WebServiceDataTransfer.java:279)
	at org.jkiss.dbeaver.model.sql.data.SQLQueryDataContainer.readData(SQLQueryDataContainer.java:156)
	at io.cloudbeaver.service.sql.WebSQLQueryDataContainer.readData(WebSQLQueryDataContainer.java:107)
	at org.jkiss.dbeaver.tools.transfer.database.DatabaseTransferProducer.lambda$0(DatabaseTransferProducer.java:271)
	at org.jkiss.dbeaver.model.exec.DBExecUtils.tryExecuteRecover(DBExecUtils.java:192)
	at org.jkiss.dbeaver.tools.transfer.database.DatabaseTransferProducer.transferData(DatabaseTransferProducer.java:188)
	at io.cloudbeaver.service.data.transfer.impl.WebServiceDataTransfer.exportData(WebServiceDataTransfer.java:332)
	at io.cloudbeaver.service.data.transfer.impl.WebServiceDataTransfer$1.run(WebServiceDataTransfer.java:186)
	at io.cloudbeaver.model.session.WebSession$1.run(WebSession.java:567)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:119)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
Caused by: java.lang.IllegalArgumentException: Invalid char ([) found at index (0) in sheet name '[select 1]'
	at org.apache.poi.ss.util.WorkbookUtil.validateSheetName(WorkbookUtil.java:150)
	at org.apache.poi.xssf.usermodel.XSSFWorkbook.createSheet(XSSFWorkbook.java:918)
	at org.apache.poi.xssf.streaming.SXSSFWorkbook.createSheet(SXSSFWorkbook.java:712)
	at org.jkiss.dbeaver.data.office.export.DataExporterXLSX.createSheet(DataExporterXLSX.java:401)
	at org.jkiss.dbeaver.data.office.export.DataExporterXLSX.getWsh(DataExporterXLSX.java:412)
	at org.jkiss.dbeaver.data.office.export.DataExporterXLSX.exportRow(DataExporterXLSX.java:427)
	at org.jkiss.dbeaver.tools.transfer.stream.StreamTransferConsumer.fetchRow(StreamTransferConsumer.java:242)
	... 11 more

--- a/plugins/org.jkiss.dbeaver.data.office/src/org/jkiss/dbeaver/data/office/export/DataExporterXLSX.java
+++ b/plugins/org.jkiss.dbeaver.data.office/src/org/jkiss/dbeaver/data/office/export/DataExporterXLSX.java
@@ -107,7 +107,7 @@ public class DataExporterXLSX extends StreamExporterAbstract implements IAppenda
     private boolean exportSql = false;
     private boolean splitSqlText = false;
     private AppendStrategy appendStrategy = AppendStrategy.CREATE_NEW_SHEETS;
-    private String exportTableName = "Sheet0";
+    private String exportTableName = XlsxSheetNameValidator.DEFAULT_SPREAD_SHEET;
 
     private int splitByRowCount = EXCEL2007MAXROWS;
     private int splitByCol = 0;
@@ -398,6 +398,7 @@ public class DataExporterXLSX extends StreamExporterAbstract implements IAppenda
             sheet = wb.getSheetAt(sheetIndex++);
             worksheet = new Worksheet(sheet, colValue, getPhysicalNumberOfRows(sheet));
         } else {
+            exportTableName = XlsxSheetNameValidator.toValidExcelSheetName(exportTableName);
             sheet = wb.createSheet(exportTableName);
             worksheet = new Worksheet(sheet, colValue, 0);
         }

--- a/plugins/org.jkiss.dbeaver.data.office/src/org/jkiss/dbeaver/data/office/export/XlsxSheetNameValidator.java
+++ b/plugins/org.jkiss.dbeaver.data.office/src/org/jkiss/dbeaver/data/office/export/XlsxSheetNameValidator.java
@@ -1,0 +1,38 @@
+/*
+ * DBeaver - Universal Database Manager
+ * Copyright (C) 2010-2024 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.data.office.export;
+
+import java.util.Set;
+
+public class XlsxSheetNameValidator {
+
+    public static final String DEFAULT_SPREAD_SHEET = "Sheet1";
+    private static final Set<String> RESERVED_WORDS = Set.of("History");
+
+    public static String toValidExcelSheetName(String originSheetName) {
+        String result = originSheetName.length() > 31 ? originSheetName.substring(0, 31) : originSheetName;
+        result = result.replaceAll("[\\\\/*\\[\\]:?]", "_");
+        result = result.strip();
+        if (result.startsWith("'") || result.endsWith("'")) {
+            result = result.replaceAll("^'+|'+$", "");
+        }
+        if (RESERVED_WORDS.contains(result) || result.isBlank()) {
+            result = DEFAULT_SPREAD_SHEET;
+        }
+        return result;
+    }
+}

--- a/plugins/org.jkiss.dbeaver.data.office/src/org/jkiss/dbeaver/data/office/export/XlsxSheetNameValidator.java
+++ b/plugins/org.jkiss.dbeaver.data.office/src/org/jkiss/dbeaver/data/office/export/XlsxSheetNameValidator.java
@@ -16,6 +16,9 @@
  */
 package org.jkiss.dbeaver.data.office.export;
 
+import org.jkiss.code.NotNull;
+import org.jkiss.code.Nullable;
+
 import java.util.Set;
 
 public class XlsxSheetNameValidator {
@@ -23,7 +26,11 @@ public class XlsxSheetNameValidator {
     public static final String DEFAULT_SPREAD_SHEET = "Sheet1";
     private static final Set<String> RESERVED_WORDS = Set.of("History");
 
-    public static String toValidExcelSheetName(String originSheetName) {
+    @NotNull
+    public static String toValidExcelSheetName(@Nullable String originSheetName) {
+        if (originSheetName == null) {
+            return DEFAULT_SPREAD_SHEET;
+        }
         String result = originSheetName.length() > 31 ? originSheetName.substring(0, 31) : originSheetName;
         result = result.replaceAll("[\\\\/*\\[\\]:?]", "_");
         result = result.strip();

--- a/plugins/org.jkiss.dbeaver.data.office/src/org/jkiss/dbeaver/data/office/export/XlsxSheetNameValidator.java
+++ b/plugins/org.jkiss.dbeaver.data.office/src/org/jkiss/dbeaver/data/office/export/XlsxSheetNameValidator.java
@@ -21,6 +21,10 @@ import org.jkiss.code.Nullable;
 
 import java.util.Set;
 
+/**
+ * Xlsx sheet name validator.
+ * Follow the rules https://support.microsoft.com/en-us/office/rename-a-worksheet-3f1f7148-ee83-404d-8ef0-9ff99fbad1f9
+ */
 public class XlsxSheetNameValidator {
 
     public static final String DEFAULT_SPREAD_SHEET = "Sheet1";

